### DISCUSSION TITLE
Windows IoT Enterprise - Soft real time also for C# ?

### DISCUSSION THREAD

**Discussion by vcarlucci** [dotnet/runtime#78497] [General] [answered]
Hi,
We are looking if we can change our architecture of our high performance environment to windows. The application is written in C#. We found the windows feature "soft real time" what is exactly the solution we are looking for, and which would allow us to dramatically changing our architecture and port all to windows.

Here the ms windows description: [ms link](https://learn.microsoft.com/en-us/windows/iot/iot-enterprise/soft-real-time/soft-real-time)

**Question**
The [code examples](https://learn.microsoft.com/en-us/windows/iot/iot-enterprise/soft-real-time/soft-real-time-application) under point 1./2. for application development showing only C++ solutions. In our NET 7 C# we found a task priority like this:

```
// C# example:
using(System.Diagnostics.Process p = Process.GetCurrentProcess())
  p.PriorityClass = ProcessPriorityClass.RealTime;
```

Now, is this "windows soft real time" supporting this C# application with such a code here? Is this equivalent and supported like in C++?

**Comment by hez2010** [comment, marked as answer]
PriorityClass wraps Win32 API directly on Windows, so it should have no difference than using Process/Thread with C++:

https://github.com/dotnet/runtime/blob/57b286fe79f0121cf87b78d8a5bcb0a9b8f6b3f8/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs#L318

**Reply by vcarlucci** [reply]
Thank you Steve for the answer and the background information. That makes sense.

**Comment by vcarlucci** [comment]
Hello @hez2010 ,
I have an issue opened related to the same topic [here](https://github.com/MicrosoftDocs/windows-iot-public/issues/11). Do you have any idea or do you can support us with that one? You can also move the issue, I was not sure where to put it in. 

Thank you

**Comment by lehar** [comment]
@vcarlucci Have you had success in getting C# to run in soft real-time? I need to create a system where I require C# to be used for soft real-time purposes. I have tried, but I haven't been able to make it run more real-time, whether I activate it on a core or not.

**Reply by snikeguo** [reply]
you need: .net aot+arm32/thumb2+rtos (like:freertos)
